BanchoSharp/Multiplayer/MultiplayerLobby.cs

```diff
@@ -390,21 +390,25 @@ private void UpdateLobbyFromBanchoBotSettingsResponse(string banchoBotResponse)
 		}
 		else if (banchoBotResponse.StartsWith("Slot "))
 		{
+			// Sample string: "Slot 1  Not Ready https://osu.ppy.sh/u/00000000 Player      [Host / HardRock]"
+			
+			// Find the digit(s) within the first 8 characters, which is the slot number
 			var slot = int.Parse((banchoBotResponse[..8].Where(c => char.IsDigit(c)).ToArray()));
-			var state = banchoBotResponse.Substring(8, 10).TrimEnd();
 
 			// Find the first ' ' after the URL, since the URL is not padded with any spaces.
 			var playerNameBegin = banchoBotResponse.IndexOf(' ', banchoBotResponse.IndexOf("/u/")) + 1;
+			
+			var playerName = banchoBotResponse.Substring(playerNameBegin, 16).TrimEnd();
+			
+			// Bancho may send extra player info after the name, for example "[Host / HardRock]", after the 16
+			// character player name bit.
+			var playerInfo = banchoBotResponse.Length > (playerNameBegin + 16) ? banchoBotResponse[(playerNameBegin + 16)..] : null;
```

Comment:
Very interesting, so BanchoBot sends 16 characters and if the name is shorter than that, the rest is filled with spaces? e.g. `Stage           ` (5 chars + 11 spaces) and `sixteensixteen  ` (14 chars + 2 spaces)

Reply:
Github doesn't show all of the white spaces I guess lol

Reply:
Correct, which makes parsing the rest easier, since we can't really trust any name to not contain any character or something.

Reply:
How does banchobot handle longer names vs those with shorter names? (e.g. the 17 char one we talked about)

Reply:
No matter - we will figure this out in testing.